crates/turborepo-lib/src/query/file.rs

```diff
@@ -0,0 +1,58 @@
+use std::sync::Arc;
+
+use async_graphql::Object;
+use itertools::Itertools;
+use turbo_trace::Tracer;
+use turbopath::AbsoluteSystemPathBuf;
+
+use crate::{query::Error, run::Run};
+
+pub struct File {
+    run: Arc<Run>,
+    path: AbsoluteSystemPathBuf,
+}
+
+impl File {
+    pub fn new(run: Arc<Run>, path: AbsoluteSystemPathBuf) -> Self {
```

Comment:
Not blocking, but would it make sense to just pass in `repo_root` instead of the entire Run object? It would make writing unit tests a lot easier.

Reply:
Eventually I want to be able to map from a file to affected packages, so that will require the whole run object